src/appleseed/renderer/kernel/rendering/pixelrendererbase.h

```diff
@@ -74,18 +78,48 @@ class PixelRendererBase
 
   protected:
     void on_pixel_begin(
-        const foundation::Vector2i& pi,
-        AOVAccumulatorContainer&    aov_accumulators);
+        const foundation::Vector2i&         pi,
+        const foundation::Vector2i&         pt,
+        const foundation::AABB2i&           tile_bbox,
+        AOVAccumulatorContainer&            aov_accumulators);
 
     void on_pixel_end(
-        const foundation::Vector2i& pi,
-        AOVAccumulatorContainer&    aov_accumulators);
+        const foundation::Vector2i&         pi,
+        const foundation::Vector2i&         pt,
+        const foundation::AABB2i&           tile_bbox,
+        AOVAccumulatorContainer&            aov_accumulators);
 
     void signal_invalid_sample();
 
   private:
-    size_t                          m_invalid_sample_count;
-    size_t                          m_invalid_pixel_count;
+    struct Parameters
+    {
+        const bool                  m_diagnostics;
```

Comment:
It would look _nicer_ to forget about trying to align this to anything. There's no hard rule here, we're clearly in the domain of aesthetics :)